From 77741a567fb88dd4d0bb9cad63912b86a5d59925 Mon Sep 17 00:00:00 2001
From: Eric Ernst <eric.ernst@intel.com>
Date: Thu, 4 Sep 2014 13:09:43 -0700
Subject: [PATCH 1/2] Add probe_slot functionality to sdhci-acpi

Adding probe_slot functionality, similar to what is available
for PCI.

Change-Id: I9faf011a3c518d991d9759c2e2fdc61150f356b1
Signed-off-by: Eric Ernst <eric.ernst@intel.com>
---
 drivers/mmc/host/sdhci-acpi.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/mmc/host/sdhci-acpi.c b/drivers/mmc/host/sdhci-acpi.c
index 6350575a093d..4cf366ed8ec5 100644
--- a/drivers/mmc/host/sdhci-acpi.c
+++ b/drivers/mmc/host/sdhci-acpi.c
@@ -67,6 +67,7 @@ struct sdhci_acpi_slot {
 	unsigned int	caps2;
 	mmc_pm_flag_t	pm_caps;
 	unsigned int	flags;
+	int (*probe_slot) (struct platform_device *);
 };
 
 struct sdhci_acpi_host {
@@ -312,6 +313,11 @@ static int sdhci_acpi_probe(struct platform_device *pdev)
 	}
 
 	if (c->slot) {
+		if (c->slot->probe_slot) {
+			err = c->slot->probe_slot(pdev);
+			if (err)
+				goto err_free;
+		}
 		if (c->slot->chip) {
 			host->ops            = c->slot->chip->ops;
 			host->quirks        |= c->slot->chip->quirks;
-- 
1.7.9.5

